src/overlays/actors/ovl_En_Fz/z_en_fz.c

```diff
@@ -26,73 +70,812 @@ const ActorInit En_Fz_InitVars = {
+    this->actor.speedXZ = 0.0f;
+    Actor_ChangeType(globalCtx, &globalCtx->actorCtx, &this->actor, ACTORTYPE_PROP);
+    Item_DropCollectibleRandom(globalCtx, &this->actor, &this->actor.posRot, 0x60);
+    this->actionFunc = EnFz_Despawn;
+}
+
+void EnFz_Despawn(EnFz* this, GlobalContext* globalCtx) {
+    if (this->timer == 0) {
+        Actor_Kill(&this->actor);
+    }
+}
+
+void EnFz_BeginMelt(EnFz* this) {
+    this->state = 3;
+    this->isFreezing = false;
+    this->isDespawning = true;
+    this->actor.flags &= ~1;
+    this->actionFunc = EnFz_Melt;
+    this->actor.speedXZ = 0.0f;
+    this->xzSpeed = 0.0f;
+}
+
+void EnFz_Melt(EnFz* this, GlobalContext* globalCtx) {
+    Math_ApproxF(&this->actor.scale.y, 0.0006f, 0.0002f);
+
+    if (this->actor.scale.y < 0.006f) {
+        this->actor.scale.x += 0.0004f;
+        this->actor.scale.z += 0.0004f;
+    }
+
+    if (this->actor.scale.y < 0.004f) {
+        this->envAlpha -= 8;
+        if (this->envAlpha >= 256) {
+            this->envAlpha = 0;
+        }
+    }
+
+    if (this->envAlpha == 0) {
+        EnFz_BeginDespawn(this, globalCtx);
+    }
+}
+
+void EnFz_BeginFreezingStationary(EnFz* this) {
+    this->state = 1;
+    this->timer = 40;
+    this->isAlwaysTrue = true;
+    this->isFreezing = true;
+    this->actor.flags |= 1;
+    this->actionFunc = EnFz_FreezingStationary;
+    this->actor.gravity = -1.0f;
+}
+
+void EnFz_FreezingStationary(EnFz* this, GlobalContext* globalCtx) {
+    Vec3f vec1;
+    Vec3f pos;
+    Vec3f velocity;
+    Vec3f accel;
+    u8 isTimerMod8;
+    s16 primAlpha;
+    s32 counterMod64;
+
+    if (this->counter & 0xC0) {
+        EnFz_RotateTowardsLink(this);
+        EnFz_GetCollision(this, globalCtx);
+    } else {
+
+        isTimerMod8 = false;
+        primAlpha = 150;
+        func_8002F974(&this->actor, NA_SE_EN_FREEZAD_BREATH - SFX_FLAG);
+
+        counterMod64 = this->counter & 0x3F; 
+        if (counterMod64 >= 48) {
+            primAlpha = 630 - (counterMod64 * 10); 
+        }
+
+        accel.z = 0.0f;
+        accel.x = 0.0f;
+        accel.y = 0.6f;
+
+        pos.x = this->actor.posRot.pos.x;
+        pos.y = this->actor.posRot.pos.y + 20.0f;
+        pos.z = this->actor.posRot.pos.z;
+
+        Matrix_RotateY((this->actor.shape.rot.y / 32768.0f) * M_PI, MTXMODE_NEW);
+
+        vec1.x = 0.0f;
+        vec1.y = -2.0f;
+        vec1.z = 20.0f;
+        Matrix_MultVec3f(&vec1, &velocity);
+
+        if ((this->counter % 8) == 0) {
+            isTimerMod8 = true;
+        }
+
+        EnFz_SetIceSmokeFreezing(this, &pos, &velocity, &accel, 2.0f, 25.0f, primAlpha, isTimerMod8);
+        pos.x += (velocity.x * 0.5f);
+        pos.y += (velocity.y * 0.5f);
+        pos.z += (velocity.z * 0.5f);
+        EnFz_SetIceSmokeFreezing(this, &pos, &velocity, &accel, 2.0f, 25.0f, primAlpha, false);
+    }
+}
+
+void EnFz_Update(Actor* thisx, GlobalContext* globalCtx) {
+    static EnFzSpawnIceSmokeFunc sSpawnIceSmoke[] = {
```

Comment:
```suggestion
    static EnFzSpawnIceSmokeFunc spawnIceSmoke[] = {
```

static arrays inside functions don't need the `s` prefix.